go/obscuronode/host/p2p.go

```diff
@@ -0,0 +1,176 @@
+package host
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+
+	"github.com/obscuronet/obscuro-playground/go/log"
+
+	"github.com/ethereum/go-ethereum/rlp"
+
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/nodecommon"
+)
+
+// TODO - Provide configurable timeouts on P2P connections.
+
+// P2P manages P2P communication between L2 nodes.
+type P2P interface {
+	// Starts listening for transaction and rollup P2P connections.
+	listen(chan nodecommon.EncryptedTx, chan obscurocommon.EncodedRollup)
+	// Stops listening.
+	stopListening()
+
+	// Broadcasts a transaction to all network peers over P2P.
+	broadcastTx([]byte)
+	// Broadcasts a rollup to all network peers over P2P.
+	broadcastRollup([]byte)
+}
+
+// NewP2P returns a new P2P object.
+// allTxAddresses is a list of all the transaction P2P addresses on the network, possibly including out own.
```

Comment:
by transaction P2P address and rollup P2P address, you mean that there are different endpoints for transactions and blocks(rollups)?
Is that how an ethereum node works?

Reply:
@tudor-malene Yes, two different endpoints. I believe Geth has a single one. But two is much simpler - you don't have to package up the bytes with some header saying what the bytes represent. You just know that bytes coming in on one address are txs, and the other rollups.

If we didn't go this route, I'd have to define some header explaining what's in the incoming packets.

Reply:
fair enough. Just make a note to have this explicit and say it's a temporary solution.

Reply:
Ok. Pedro wants me to consolidate so I'll probably do that.